src/main/java/hudson/tasks/Mailer.java

```diff
@@ -343,17 +348,20 @@ public void setReplyToAddress(String address) {
          * @return mail session based on the underlying session parameters.
          */
         public Session createSession() {
-            return createSession(smtpHost,smtpPort,useSsl,getSmtpAuthUserName(),getSmtpAuthPasswordSecret());
+            return createSession(smtpHost,smtpPort,useSsl,useTls,getSmtpAuthUserName(),getSmtpAuthPasswordSecret());
         }
-        private static Session createSession(String smtpHost, String smtpPort, boolean useSsl, String smtpAuthUserName, Secret smtpAuthPassword) {
+        private static Session createSession(String smtpHost, String smtpPort, boolean useSsl, boolean useTls, String smtpAuthUserName, Secret smtpAuthPassword) {
+            final String SMTP_PORT_PROPERTY = "mail.smtp.port";
+            final String SMTP_SOCKETFACTORY_PORT_PROPERTY = "mail.smtp.socketFactory.port";
+
             smtpPort = fixEmptyAndTrim(smtpPort);
             smtpAuthUserName = fixEmptyAndTrim(smtpAuthUserName);
 
             Properties props = new Properties(System.getProperties());
             if(fixEmptyAndTrim(smtpHost)!=null)
                 props.put("mail.smtp.host",smtpHost);
```

Comment:
as you introduce the `SMTP_PORT_PROPERTY` variable, would be nice to use it here as well probably.

```suggestion
                props.put(SMTP_PORT_PROPERTY, smtpHost);
```

Reply:
Port is different from host :/

Reply:
:man_facepalming: